File: SafeCityAPI/src/main/java/com/ph/model/Incident.java
package com.ph.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.geo.GeoJsonPoint;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.CompoundIndexes;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.util.Date;

/**
 * Created by leon on 25/05/2017.
 */
@Document(collection = "incident")
@CompoundIndexes({
        @CompoundIndex(name = "geolocation", def = "{'location': '2dsphere'}")
})
public class Incident implements Serializable{

    private static final long serialVersionUID = 1L;

    @Id
    private String _id;

    private GeoJsonPoint location;
    private String title;
    private String description;
    private String incident;
    private Date date;
    private int level;

    public Incident(GeoJsonPoint location, String title, String description, String incident, int level, Date date) {
        this.location = location;
        this.title = title;
        this.description = description;
        this.incident = incident;
        this.level = level;
        this.date = date;
    }

    public Incident() {
    }

    @Override
    public String toString() {
        return "Incident{" +
                "_id=" + _id.toString() +
                ", location=" + location.toString() +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", incident='" + incident + '\'' +
                ", date=" + date +
                ", level=" + level +
                '}';
    }

    public String get_id() {
        return _id;
    }

    public GeoJsonPoint getLocation() {
        return location;
    }

    public void setLocation(GeoJsonPoint location) {
        this.location = location;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getIncident() {
        return incident;
    }

    public void setIncident(String incident) {
        this.incident = incident;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public int getLevel() {
        return level;
    }

    public void setLevel(int level) {
        this.level = level;
    }
}

File: SafeCityAPI/src/main/java/com/ph/model/Incident.kt
package com.ph.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.geo.GeoJsonPoint
import org.springframework.data.mongodb.core.index.CompoundIndex
import org.springframework.data.mongodb.core.index.CompoundIndexes
import org.springframework.data.mongodb.core.mapping.Document
import java.io.Serializable
import java.util.*

/**
 * Created by leon on 11/30/2017.
 */
@Document(collection = "incident")
@CompoundIndexes(CompoundIndex(name = "geolocation", def = "{'location': '2dsphere'}"))
data class Incident constructor(
        val location: GeoJsonPoint,
        val title: String,
        val description: String,
        val incident: String,
        val level: Int,
        val date: Date) : Serializable {

    @Id
    var _id: String? = null

    override fun toString(): String {
        return "Incident{" +
                "_id=" + _id +
                ", location=" + location.toString() +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", incidentService='" + incident + '\'' +
                ", date=" + date +
                ", level=" + level +
                '}'
    }

    companion object {

        private const val serialVersionUID = 1L
    }

    fun toMap(): HashMap<String, Serializable?> {
        return hashMapOf("_id" to this._id, "location" to this.location, "title" to this.title,
                "description" to this.description, "incident" to this.incident, "date" to this.date, "level" to this.level)
    }
}